server/api.js

```diff
@@ -2,100 +2,416 @@ import { Router } from "express";
+			} else {
+				res.status(404).json({
+					status: 404,
+					msg: "This user doesn't have a goal with this id",
+				});
+			}
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//POST a goal for specific plan id
+router.post("/users/plans/:plan_id/goals", (req, res) => {
+	const { plan_id } = req.params;
+	const { title, status, start_date, end_date } = req.body;
+	const query =
+		"INSERT INTO goals ( plan_id, title, status, start_date, end_date ) VALUES ($1, $2, $3, $4, $5) RETURNING *";
+	pool
+		.query(query, [plan_id, title, status, start_date, end_date])
+		.then((result) => {
+			res.json({ success: "New plan has been created!", user: result.rows[0] });
+		})
+		.catch((err) => console.error(err.message));
+});
+
+//DELETE a goal for specific plan id
+router.delete("/users/plans/:plan_id/goals/:goal_id", (req, res) => {
+	const { plan_id, goal_id } = req.params;
+	pool
+		.query("SELECT * FROM goals WHERE plan_id = $1 and id = $2", [
+			plan_id,
+			goal_id,
+		])
+		.then(() => {
+			pool
+				.query("DELETE FROM goals WHERE id=$1", [goal_id])
+				.then(() => res.send(`goal ${goal_id} has been deleted!`))
+				.catch((e) => console.error(e));
+		})
+		.catch((err) => console.error(err.message));
+});
+
+// GET all tasks form specific goal id
+router.get(
+	"/users/:user_id/plans/:plan_id/goals/:goal_id/tasks",
+	(req, res) => {
+		const { user_id, plan_id, goal_id } = req.params;
+		const query = `SELECT t.id, t.goal_id goal_id, t.description, t.status, t.create_date  FROM tasks t 
+						INNER JOIN goals g on t.goal_id = g.id 
+						INNER JOIN plans p on g.plan_id = p.id 
+						where p.user_id =$1 and p.id= $2 and t.goal_id=$3`;
+		pool
+			.query(query, [user_id, plan_id, goal_id])
+			.then((result) => {
+				if (result.rowCount) {
+					res.json(result.rows);
+				} else {
+					res.status(404).json({
+						status: 404,
+						msg: "This user doesn't have a goal with this id",
+					});
+				}
+			})
+			.catch((err) => console.error(err.message));
+	}
 );
-res.json(newUser.rows[0]);
-} catch (err) {
-console.error(err.message);
-}
+
+// GET all feedbacks form specific user id
+router.get("/users/:user_id/feedbacks", async (req, res) => {
+	const { user_id } = req.params;
+	try {
+		const feedback_req = await pool.query(
+			`SELECT graduate.first_name, graduate.id, p.description, f.description, f.id, mentors.first_nameFROM feedbacks as f 
+				INNER JOIN plans  as p ON p.id = f.plan_id INNER JOIN users as graduate ON graduate.id = p.user_id
+				INNER JOIN users as mentors ON f.user_id= mentors.id where mentors.id = $1
+					ORDER BY f.create_date`,
+			[user_id]
+		);
+		res.json(feedback_req.rows);
+	} catch (err) {
+		res.status(500).json("server error");
+	}
 });
 
-router.get("/users/:id", async (req, res) => {
+// POST feedback form specific user id (mentor)
+router.post("/users/:user_id/feedbacks", async (req, res) => {
+	const { id } = req.body;
+	const { user_id } = req.params;
```

Comment:
Should we have `description`? What about other fields such as `created_at_date`?